isPlainObject.js

```diff
@@ -35,9 +35,9 @@ function isPlainObject(value) {
     return true
   }
   let proto = value
-  while (Object.getPrototypeOf(proto) !== null) {
+  do {
     proto = Object.getPrototypeOf(proto)
-  }
+  } while (Object.getPrototypeOf(proto) !== null)
```

Comment:
You are changing the while head check here. In the very first check, `Object.getPrototypeOf(proto)` is checked to not be `null`. But in your code, you don't check if the resulting `proto = Object.getPrototypeOf(proto)` is `null` or not.